### PR TITLE
CFY-7632 integrate the aria rest mapi into the rest client

### PR DESCRIPTION
**Pull request by mxmrlv**

### CODE REVIEW

File: cloudify_rest_client/aria/mapi/mapi.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        if isinstance(value, dict):
+            wrapper = wrappers.DictWrapper
+            kw = dict(
+                    (key, self._wrap(value, '/'.join([attribute_path, key])))
+                    for key, value in (value or {}).items()
+                )
+        elif isinstance(value, list):
+            wrapper = wrappers.ListWrapper
+            seq = [self._wrap(
+                item, '/'.join([attribute_path, str(getattr(item, 'id', i))])
```
Comment (aviyoop) [inline cloudify_rest_client/aria/mapi/mapi.py]: Check why the `id`. Why not just `i`?

File: cloudify_rest_client/aria/mapi/mapi.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            ) for i, item in enumerate(value or [])]
+            kw = dict(seq=seq)
+
+        elif isinstance(value, ModelMixin):
+            wrapper = wrappers.ObjectWrapper
+            kw = dict(obj=value)
+
+        if wrapper is not None and callable(wrapper):
```
Comment (aviyoop) [inline cloudify_rest_client/aria/mapi/mapi.py]: Just `callable`.

File: cloudify_rest_client/aria/mapi/mapi.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                _query=partial(self._get_query,
+                               end_point=self._endpoint,
+                               responder=self._respond),
+                **kw
+            )
+
+        return value
+
+    @classmethod
+    def _get_query(cls, entry_path, end_point, responder):
```
Comment (aviyoop) [inline cloudify_rest_client/aria/mapi/mapi.py]: explain what is the entry path (what is its relation to the attribute path, etc).

File: cloudify_rest_client/aria/mapi/wrappers.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        self.__dict__['_query'] = _query
+
+    def __getattr__(self, item):
+        try:
+            return self[item]
+        except KeyError:
+            return self._query('/'.join([self._attribute_path, item]))
+
+
+class ObjectWrapper(WrapperBase):
```
Comment (aviyoop) [inline cloudify_rest_client/aria/mapi/wrappers.py]: `ModelWrapper`.

File: cloudify_rest_client/aria/mapi/wrappers.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+#    * limitations under the License.
+
+
+class WrapperBase(object):
+
+    def __init__(self, _attribute_path, _query, **_):
+        self.__dict__['_attribute_path'] = _attribute_path
+        self.__dict__['_query'] = _query
+
+    def __getattr__(self, item):
```
Comment (aviyoop) [inline cloudify_rest_client/aria/mapi/wrappers.py]: Remove, no need for this.

File: cloudify_rest_client/aria/mapi/wrappers.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        return (isinstance(other, ObjectWrapper) and
+                other._obj.id == self._obj.id and
+                other._obj.__class__ == self._obj.__class__)
+
+
+class DictWrapper(dict, WrapperBase):
+
+    def __init__(self, **kwargs):
+        WrapperBase.__init__(self, **kwargs)
+        dict.__init__(self, **kwargs)
```
Comment (aviyoop) [inline cloudify_rest_client/aria/mapi/wrappers.py]: Think for a solution for the duplicity with `**kwargs`

File: cloudify_rest_client/aria/mapi/mapi.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+                              if isinstance(entry, wrappers.WrapperBase)
+                              else entry)
+        )
+        return self._respond(response, self.name)
+
+    def create(self, **kwargs):
+        pass
+
+    def get(self, entry_id, **kwargs):
+        suffix = '/'.join([self.name, str(entry_id)])
```
[the remark this answers is not in the document]
Comment (mxmrlv) [reply cloudify_rest_client/aria/mapi/mapi.py]: Oh, but it is.

File: cloudify_rest_client/aria/mapi/mapi.py [hunk 8]
```diff
@@ -93,20 +88,23 @@ def _wrap(self, value, attribute_path):
             wrapper = wrappers.DictWrapper
             kw = dict(
                     (key, self._wrap(value, '/'.join([attribute_path, key])))
-                    for key, value in (value or {}).items()
+                    for key, value in (value).items()
```
Comment (aviyoop) [inline cloudify_rest_client/aria/mapi/mapi.py]: remove parenthesis

File: cloudify_rest_client/aria/mapi/mapi.py [hunk 9]
```diff
@@ -93,20 +88,23 @@ def _wrap(self, value, attribute_path):
...
                 )
         elif isinstance(value, list):
             wrapper = wrappers.ListWrapper
             seq = [self._wrap(
-                item, '/'.join([attribute_path, str(getattr(item, 'id', i))])
-            ) for i, item in enumerate(value or [])]
+                item, '/'.join([attribute_path, str(i)])
+            ) for i, item in enumerate(value)]
+            # seq = [self._wrap(
```
Comment (aviyoop) [inline cloudify_rest_client/aria/mapi/mapi.py]: add ' # TODO ...'

File: cloudify_rest_client/aria/mapi/mapi.py [hunk 10]
```diff
@@ -119,5 +117,13 @@ def _wrap(self, value, attribute_path):
 
     @classmethod
     def _get_query(cls, entry_path, end_point, responder):
+        """
+        Enables sending generic queries to the rest service
+
+        :param entry_path: The path of the attribute.
+        :param end_point: The rest service endpoint.
+        :param responder: the responed method to wrap with
```
Comment (aviyoop) [inline cloudify_rest_client/aria/mapi/mapi.py]: The
responder

**Pull request merged**